src/app/calendar/getParcelCalendarEvents.ts

```diff
@@ -0,0 +1,83 @@
+import supabase, { Schema } from "@/supabase";
+import { CalendarEvent } from "@/components/Calendar/Calendar";
+
+const COLLECTION_DURATION_MS = 30 * 60 * 1000;
+
+const colorMap: { [location: string]: string } = {
+    "Brixton Hill - Methodist Church": "#d50300",
+    "Clapham - St Stephens Church": "#f4511e",
+    "N&B - Emmanuel Church": "#f6bf25",
+    "Streatham - Immanuel & St Andrew": "#33b679",
+    "Vauxhall Hope Church": "#0a8043",
+    "Waterloo - Oasis": "#039be5",
+    "Waterloo - St George the Martyr": "#3f51b5",
+    "Waterloo - St Johns": "#7986cb",
+    Delivery: "#8e24aa",
+    default: "#626161",
+};
+
+interface ClientMap {
+    [primary_key: string]: Schema["clients"];
+}
+
+const fetchClient: () => Promise<Schema["clients"][]> = async () => {
+    const response = await supabase.from("clients").select();
+    return response.data ?? [];
+};
+
+const fetchParcel: () => Promise<Schema["parcels"][]> = async () => {
+    const response = await supabase.from("parcels").select();
+    return response.data ?? [];
+};
+
+const clientsToClientMap = (clients: Schema["clients"][]): ClientMap => {
+    const clientMap: ClientMap = {};
+
+    clients.forEach((client) => {
+        clientMap[client.primary_key] = client;
+    });
+
+    return clientMap;
+};
+
+const getParcelsWithCollectionDate = (parcels: Schema["parcels"][]): Schema["parcels"][] => {
+    return parcels.filter((parcel) => parcel.collection_datetime !== null);
```

Comment:
this is likely better as just selecting the data we need at a database level rather than getting all the data and filtering it in a server somewhere!
look into [supabase filtering](https://supabase.com/docs/reference/javascript/using-filters) - I think doing `.select().not("collection_datetime", "is", null)` should work?

Reply:
Done